### PR TITLE
Include type in streaming TestPartResult output (fixes #3994).

### LINKED ISSUE #3994: Streaming output doesn't distinguish between SUCCESS() and failure messages
**Describe the bug**

When processing streaming output for TestPartResult, it's not possible to tell if it's an error or not.

**Steps to reproduce the bug**

```c++
TEST(SomeFixture, SampleExplicitSucceed)
{
    SUCCEED();
}

TEST(SomeFixture, SampleFails)
{
    constexpr int x{ 1 };
    constexpr int y{ 2 };
    ASSERT_EQ(x, y);
}
```

**Does the bug persist in the most recent commit?**

yes

**What operating system and version are you using?**

Windows 10 (10.0.19044.1889)

**What compiler and version are you using?**

MSVC with VS 2022 (VS 17.3.3)

**What build system are you using?**

Visual Studio
**Additional context**

The streaming output looks like this:
```
...
event=TestPartResult&file=path\to\some_file.cpp&line=15&message=Succeeded
...
event=TestPartResult&file=path\to\some_file.cpp&line=22&message=Expected equality of these values:%0A  x%0A    Which is: 1%0A  y%0A    Which is: 2
...
```

The TestPartResult contains a type field - if that were preserved in event=TestPartResult (&type=...), programmatic marking of the message as error or not would be possible.

### PR DESCRIPTION
**Pull request by davidmatson**

### CODE REVIEW

**Comment by davidmatson** [comment]
@asoffer - could you approve the workflow to run? Thanks!

File: googletest/test/gtest_unittest.cc [hunk 1]
```diff
@@ -143,7 +143,37 @@ TEST_F(StreamingListenerTest, OnTestPartResult) {
 
   // Meta characters in the failure message should be properly escaped.
   EXPECT_EQ(
-      "event=TestPartResult&file=foo.cc&line=42&message=failed%3D%0A%26%25\n",
+      "event=TestPartResult&type=failure&file=foo.cc&line=42&message=failed%3D%0A%26%25\n",
+      *output());
+}
+
+TEST_F(StreamingListenerTest, OnTestPartResultSuccess) {
+  *output() = "";
```
Comment (krishnacx) [inline googletest/test/gtest_unittest.cc:151]: Some formating required. Check previous TEST_F invocations.

Comment (davidmatson) [reply googletest/test/gtest_unittest.cc:151]: I tried to mode this one on the one above:
https://github.com/google/googletest/blob/459bd19cdf77b3c73dbdd308f08d002aaa46b1ea/googletest/test/gtest_unittest.cc#L139

I checked again and don't see a formatting difference, but maybe I'm missing something.

**Comment by davidmatson** [comment]
@asoffer - I'm hoping this is a straightforward change to review. Would you be able to take a look?

Thanks!

**Comment by davidmatson** [comment]
@krishnacx - would you be able to help with reviewing this change? Thanks!

**Comment by davidmatson** [comment]
@dinord - would you be able to help with moving this PR forward? Thanks!

**Comment by davidmatson** [comment]
It looks like this PR is stuck; abandoning.

**Pull request closed without merging**